README.md

```diff
@@ -328,6 +328,39 @@ Calculates the new alpha version, creates new git tag and publishes the new pack
     secrets: inherit
 ```
 
+## Cookbook
+
+This section contains a list of recipes and common patterns organized by desired
+outcome.
+
+### Serialize pull request builds
+
+When a workflow requires to access an external shared resource, it may be
+desirable to prevent concurrent builds of the same pull request with:
+
+```yml
+name: my-workflow
+on:
+  pull_request:
+    branches:
+      - develop
+concurrency: ${{ github.head_ref || github.run_id }}
+```
+
+or for workflows with push:
+
+```yml
+name: my-workflow
+on:
+  push:
+concurrency: ${{ github.ref_name || github.run_id }}
```

Comment:
if this a shared block, when you have both pull_request and push which is a typical case what would you set?